ci/spack.yaml

```diff
@@ -8,7 +8,6 @@ spack:
     - matrix:
       - [$packages]
       - [$%compilers]
-      - [arch=linux-ubuntu20.04-x86_64_v3]
```

Comment:
Can we keep the x86_64_v3 arch? Spack is aggressive about using the largest feature set for architecture which can include avx512 and is only supported on some github nodes. So there can be a subtle bug where the docker image created on one node can cause SIGILL on another node